lib/model.js

```diff
@@ -315,247 +215,243 @@ function _create (data, definition) {
-            .then(function (result) {
-              if (result.matchedCount === 0) {
-                throw errors.Conflict({ message: 'Document update conflict' });
-              }
-              debug('update into ' + collection + ' successful', result);
-              var updatedModel = this.set({
-                _etag: newEtag,
-                __modified: [],
-                __original: null
-              }, { silent: true });
-              if ('function' === typeof updatedModel.afterSave) {
-                updatedModel.afterSave(original);
-              }
-              return updatedModel;
-            }.bind(this));
-        } else {
-          obj._etag = newEtag;
-          debug('inserting into ' + collection + ': ' + JSON.stringify(obj));
-          return db.insert(collection, obj)
-            .then(function (result) {
-              debug('insert into ' + collection + ' successful', result);
-              var obj = Array.isArray(result.ops) ? result.ops[0] : result.ops;
-              var newModel = _create(obj, definition);
-              if ('function' === typeof newModel.afterSave) {
-                newModel.afterSave(null);
-              }
-              return newModel;
-            }.bind(this));
-        }
-      }.bind(this));
+    if ('function' === typeof newOrUpdatedModel.afterSave) {
+      newOrUpdatedModel.afterSave(original || null);
+    }
+    return newOrUpdatedModel;
   };
 
-  definition.methods.remove = function () {
-    var collection = definition.collection;
-    var query = { _id: this._id, _etag: this._etag };
-    var before = 'function' === typeof this.beforeRemove ?
-      q.fcall(() => { return this.beforeRemove(); }) : q.resolve();
-    return before.then(function () {
-      debug('removing ' + JSON.stringify(query) + ' in ' + collection);
-      return db.remove(collection, query)
-        .then(function (result) {
-          if (result.deletedCount === 0) {
-            throw errors.Conflict({ message: 'Document remove conflict' });
-          }
-          debug('remove from ' + collection + ' successful');
-          if ('function' === typeof this.afterRemove) {
-            this.afterRemove();
-          }
-          return result;
-        }.bind(this));
-    }.bind(this));
+  definition.methods.remove = async function() {
+    const collection = definition.collection;
+    const query = { _id: this._id, _etag: this._etag };
+    const before = async () => {
+      if ('function' === typeof this.beforeRemove) {
+        return this.beforeRemove();
+      }
+    };
+    await before();
+    debug(`removing ${JSON.stringify(query)} in ${collection}`);
+    const result = await db.remove(collection, query);
+    if (result.deletedCount === 0) {
+      throw errors.Conflict({ message: 'Document remove conflict' });
+    }
+    debug(`remove from ${collection} successful`);
+    if ('function' === typeof this.afterRemove) {
+      // should this be awaited??
```

Comment:
think we decided on no, and should probably have an explicit comment to that effect?